File: .cursor/state/workflow-state.py
#!/usr/bin/env python3
"""Compatibility shim for the workflow-state API and CLI.

The canonical implementation lives under
``src/oh_my_cursor/workflow_state/``. This checked-in shim remains so older
installed plugin payloads and direct developer commands keep working while
``.cursor/state/`` stays the schema/runtime contract surface.
"""
from __future__ import annotations

import importlib
import sys
from pathlib import Path

_SRC_DIR = Path(__file__).resolve().parents[2] / "src"
if str(_SRC_DIR) not in sys.path:
    sys.path.insert(0, str(_SRC_DIR))

_api = importlib.import_module("oh_my_cursor.workflow_state.api")
_cli = importlib.import_module("oh_my_cursor.workflow_state.cli")

AC_STATUSES = getattr(_api, "AC_STATUSES")
DEFAULT_HISTORY_CAP = getattr(_api, "DEFAULT_HISTORY_CAP")
FAILURE_TYPES = getattr(_api, "FAILURE_TYPES")
PHASES = getattr(_api, "PHASES")
ROLES = getattr(_api, "ROLES")
STATUSES = getattr(_api, "STATUSES")
append_history = getattr(_api, "append_history")
build_parser = getattr(_cli, "build_parser")
file_lock = getattr(_api, "file_lock")
init_state = getattr(_api, "init_state")
main = getattr(_cli, "main")
read_state = getattr(_api, "read_state")
record_failure = getattr(_api, "record_failure")
set_state = getattr(_api, "set_state")
update_acceptance_criterion = getattr(_api, "update_acceptance_criterion")

__all__ = [
    "AC_STATUSES",
    "DEFAULT_HISTORY_CAP",
    "FAILURE_TYPES",
    "PHASES",
    "ROLES",
    "STATUSES",
    "append_history",
    "build_parser",
    "file_lock",
    "init_state",
    "main",
    "read_state",
    "record_failure",
    "set_state",
    "update_acceptance_criterion",
]


if __name__ == "__main__":
    raise SystemExit(main())

File: .cursor/state/workflow-state.ts
import * as process from 'node:process';
import * as path from 'node:path';
import { fileURLToPath } from 'node:url';
import * as api from '../../src/oh_my_cursor/workflow_state/api.ts';
import { fileLock } from '../../src/oh_my_cursor/workflow_state/locking.ts';
import { main as cliMain } from '../../src/oh_my_cursor/workflow_state/cli.ts';

export const DEFAULT_HISTORY_CAP = api.DEFAULT_HISTORY_CAP;
export const PHASES = api.PHASES;
export const STATUSES = api.STATUSES;
export const ROLES = api.ROLES;
export const AC_STATUSES = api.AC_STATUSES;
export const FAILURE_TYPES = api.FAILURE_TYPES;

export const read_state = api.readState;
export const init_state = api.initState;
export const set_state = api.setState;
export const update_acceptance_criterion = api.updateAcceptanceCriterion;
export const record_failure = api.recordFailure;
export const append_history = api.appendHistory;
export const file_lock = fileLock;
export const main = cliMain;

const currentFile = fileURLToPath(import.meta.url);
const isDirectRun = process.argv[1] && path.resolve(process.argv[1]) === path.resolve(currentFile);
if (isDirectRun) {
  cliMain();
}
